Initial reusable workflow implementation

charles2910 opened the pull request:
This reusable workflow builds on top of other actions and act as the interface for them. In this way, every change in the workflow is done in a unique and centralized place instead of MRing many repos.

Initial options allowed are:
- build package;
- publish artifacts;
- attest artifacts;
- run piuparts.

charles2910 commented:
Should I put someone else as reviewer?

leonheldattoradex commented:
@charles2910, this looks good but I'd like it to be squashed into one commit. Having these many commits for an already very readable file is not very good for commit history.

Commit:
toradex-deb-ci: add initial implementation of reusable workflow

Current supports build image, upload artifacts, attest build artifacts
and run piuparts.

Related-to: TCCP-650, TCCP-750
Signed-off-by: Carlos Henrique Lima Melara <carlos.melara@toradex.com>

leonheldattoradex reviewed (approved):
Thanks!
